### ISSUE TITLE
TypeError: overrideId is not a function

### ISSUE THREAD

**Issue by sunupf** [annotorious/annotorious#189]
Hi, based on the API docs, there are 2 params on function callback

1. annotation
2. overrideId

I was trying to use overrideId and it give `TypeError: overrideId is not a function` . here my code

```
this.anno.on('createAnnotation', async function(annotation, overrideId) {
    const slideId = self.slide.id
    const savedAnnotation = await self.saveAnnotation(slideId, annotation)
    overrideId(savedAnnotation.id)
});
```

I use @recogito/annotorious-openseadragon: 2.6.0 which using annotorius v 2.6.0 and client core 1.6.0

Is it my implementation is wrong or something else?

Thanks in advance

**Comment by rsimon** [comment]
Hi @sunupf,

weird - your code looks ok. I just checked & can't seem to reproduce the problem. OverrideId is correctly passed to the `createAnnotation` event handler, both in AnnotoriousOSD 2.6 and the latest 2.7. I also tried with an async/await example and that seemed to work just fine, too.

Perhaps there's something additional going on in your code. Do you have a full code sample you can post? Can you try to log the value of `overrideId` to see if something else is being passed?

**Comment by sunupf** [comment]
Hi thanks for the quick reply,

ok let me explain my code.

1. We user headless mode, so we can show custom popup to input label, (why not using widget, well I guess we try something that comfort for my team :)) . it's something like this :
<img width="290" alt="Screen Shot 2022-01-05 at 19 56 18" src="https://user-images.githubusercontent.com/6140252/148221120-c69b6ecc-ca04-4acf-83e7-784a5b124ba9.png">

2. When user insert the label, we update the annotation using
```
await this.anno.updateSelected(this.selectedAnno);
this.anno.saveSelected();
```

3. Because it will trigger `createAnnotataion` event, I code the save and try to override the Id using previous code
```
this.anno.on('createAnnotation', async function(annotation, overrideId) {
    const slideId = self.slide.id
    const savedAnnotation = await self.saveAnnotation(slideId, annotation)
    overrideId(savedAnnotation.id)
});
```

but it throw error when the code hit `overrideId`. Here's the screenshoot for error an I try console.log before the call
<img width="383" alt="Screen Shot 2022-01-05 at 19 59 33" src="https://user-images.githubusercontent.com/6140252/148221588-6efcde9a-c2a3-4c8a-b591-e59eb9a4224c.png">

FYI : I use annotorious plugin called `@recogito/annotorious-selector-pack` , it seems use different version of annotorious  
<img width="564" alt="Screen Shot 2022-01-05 at 20 02 02" src="https://user-images.githubusercontent.com/6140252/148221794-376a35f9-ed27-4765-a2ee-6e0d6b24a38d.png">

Thanks
Sunu

**Comment by rsimon** [comment]
Thanks for this! Yikes - that is indeed a bug in Annotorious. Works in normal mode (where I tried), but is broken in headless mode. I have already added a fix to the main branch in [annotorious-openseadragon](https://github.com/recogito/annotorious-openseadragon). Will do a bit more testing & check if the same problem also exists in Annotorious' normal version (non-OSD). The next release will definitely work.
